src/engine/renderer/gl_shader.cpp

```diff
@@ -1582,45 +1582,19 @@ std::string GLShaderManager::ShaderPostProcess( GLShader *shader, const std::str
 	* #define uniformx materials[baseInstance].uniformx
 	*/
 
-	for( GLUniform* uniform : shader->_uniforms ) {
-		if ( uniform->IsGlobal() ) {
-			continue;
-		}
-
-		if ( !uniform->IsTexture() ) {
-			materialStruct += "	" + uniform->GetType() + " " + uniform->GetName();
-
-			if ( uniform->GetComponentSize() ) {
-				materialStruct += "[ " + std::to_string( uniform->GetComponentSize() ) + " ]";
-			}
-			materialStruct += ";\n";
-
-			// vec3 is aligned to 4 components, so just pad it with int
-			// TODO: Try to move 1 component uniforms here to avoid wasting memory
-			if ( uniform->GetSTD430Size() == 3 ) {
-				materialStruct += "	int ";
-				materialStruct += uniform->GetName();
-				materialStruct += "_padding;\n";
```

Comment:
I don't see any replacement for this. Does that mean it is going to break if there aren't enough 4-byte uniforms to fill all the vec3 holes?

Reply:
But `WriteToBuffer` is now assuming that there is never padding.